notebooks/02_model/rbm_deep_dive.ipynb

```diff
@@ -0,0 +1,1449 @@
+    "\n",
+    "The RBM approach is to look at $X$ as a particular realization (sample) of a more general process; instead of learning a specific $X$, we try to learn the matrix distrbution from which $X$ has been sampled form. Efectively, we learn the typical distirbution of *tastes* (i.e. latent factors) and use this information to *generate* new ratings. For this reaoson, this class of neural network models is also called **generative**. Consider the following example: imagine you are given the  income distribution per age window of a particular country (this is easy to find from goverments data), then we could fix the age window and *generate* virtual citizens with various incomes by sampling from this distibution.   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1.2 Model \n",
+    "\n",
+    "The central quantity of every SM model is the [Boltzmann distribution ](https://en.wikipedia.org/wiki/Boltzmann_distribution); this can be seen as the least biased probability distribution on a given probability space $\\Sigma$ and can be obtained using a maximum entropy principle on the space of distributions over $\\Sigma$. Its typical form is: \n",
+    "\n",
+    "$P = \\frac{1}{Z} \\, e^{- \\beta \\, H}$, \n",
+    "\n",
+    "where $Z$ is a normalization constant known as the partition function, $\\beta$ is a noise parameter with units of inverse energy and $H$ is the Hamiltonian, or energy function of the system. For this reason, this class of models is also known as *energy based* in computer science. In physics, $\\beta$ is the inverse temperature of the system in units of Boltzmann's constant, but here we will effectively rescale inside $H$, so that this is now a pure number. $H$ describes the behaviour of two sets of stochastic vectors, typically called $v_i$ (visibles) and $h_j$ (hidden). The former constitutes both the input *and* the ouput of the algo (this will be clear later), while the hidden units are the latent factors we want to learn. This structure results in the following Neural Network topology\n",
+    "\n",
+    "![rbm1](https://393229b54263dsvm.file.core.windows.net/images/RBM1.png?sp=rl&st=2018-12-03T13:06:28Z&se=2020-12-04T13:06:00Z&sv=2017-11-09&sig=XP5y%2BXuU4r3Snmb%2FIgksMR%2B26XZfgrV2ZVs183Nwn3k%3D&sr=f)\n",
```

Comment:
Image is not rendered well.

Reply:
Aaron added an image folder in 02_model: https://github.com/Microsoft/Recommenders/tree/staging/notebooks/02_model/img. One option would be to put it there. 

Another option would be to host all the images in a blob and link the images from there.